README.md

```diff
@@ -59,6 +60,23 @@ services.AddSitemaps(x =>
 }, p => p.RequireRole(Roles.Administrators));
 ```
 
+In order to augment Urls for the PersonListPage with the corresponding querystring parameters for said page, please review the [SitemapUriParameterAugmenterService class](sandbox/Foundation/src/Foundation/Infrastructure/Cms/Services/SitemapUriParameterAugmenterService.cs) within the Foundation project:
+
+```csharp
+services.AddSitemaps(uriAugmenterService: sp => sp.GetInstance<SitemapUriParameterAugmenterService>());
+```
+
+Or, alternatively:
+```csharp
+services.AddSitemaps(x =>
+{
+  x.EnableLanguageDropDownInAdmin = false;
+  x.EnableRealtimeCaching = true;
+  x.EnableRealtimeSitemap = false;
+}, sp => sp.GetInstance<SitemapUriParameterAugmenterService>());

```

Comment:
That's not how I imagined setting it. 
It should be something like:
```suggestion
services.AddSitemaps(x =>
{
  x.EnableLanguageDropDownInAdmin = false;
  x.EnableRealtimeCaching = true;
  x.EnableRealtimeSitemap = false;
  x.SetAugmenterService<SitemapUriParameterAugmenterService>();
});
```
The options class should have a property with an augmenter service type, it will register in a container. By default, it should be the DefaultAugmenterService, with `SetAugmenterService` call it should replace that type.
Then in the ServiceCollectionExtensions, register that service by type set on the property.
Here is an example of Options class: 
https://github.com/Geta/geta-optimizely-productfeed/blob/master/src/Geta.Optimizely.ProductFeed/Configuration/ProductFeedOptions.cs
And here is how it looks like in a ServiceCollectionExtensions:
https://github.com/Geta/geta-optimizely-productfeed/blob/master/src/Geta.Optimizely.ProductFeed/ServiceCollectionExtensions.cs#L89
It will be simpler for AugmenterService though.

Reply:
@marisks that's what I was trying previously. However, the optionsAction was not being called at all. It appears to be a bug to me, which is why I wrote it the way I did. I'm trying to wrap things up here basically as I'm at my time.

Reply:
It is firing. What exactly didn't work?

Reply:
I could try it again, but I couldn't hit a break point inside of the Options and the code was erroring saying that the service had not been configured. I can try again if you like. I'll check it in and show you what I've done and the issue I'm having.

Reply:
@jeff-fischer-optimizely The service will not automatically register, you have to implement it as mentioned in my original comment - see the links on how it is implemented in Geta.Optimizely.ProductFeed.

Reply:
You're right, I have implemented it slightly different, mainly that I'm passing in IServiceCollection into the Set method. But, the issue I was trying to point out - that the optionsAction is not being called - still stands true. Whether I implement it on your way or mine, the optionsAction will not be executed and the code won't work.

Please try debugging my code and you'll see the optionsAction is not being called.

I've set a breakpoint within optionsAction and it doesn't get called within the Foundation example project.

Reply:
Oh, I see. You're actually calling the optionsAction yourself. That will work since it's being called directly.

But, FYI, this line of code is not being called.

![image](https://user-images.githubusercontent.com/91963938/193850260-8e735480-b277-4e8e-8a6c-0a7b6aa4edab.png)

Reply:
IMO, it's a little hackish to call the optionsAction explicitly. My preference would be to tie it into DI and only call the optionsAction once instead of twice.

Reply:
@jeff-fischer-optimizely That line is called when options are injected into some service.
But you must call it twice - once before registering augmenter service and the second time here. Otherwise, it will not populate values correctly.

Reply:
Yes, it is hackish, but I haven't found a better solution yet as DI has not been built yet, I cannot request service from it.

Reply:
Yeah, I see. Silly me, I was thinking of doing where setupAction gets called (which it doesn't). But, that's after the container is already built. I should have considered that.

Thanks, I've updated the code based on your suggestions and I've just tested it successfully.